clients/src/main/java/org/apache/kafka/clients/NetworkClient.java

```diff
@@ -680,6 +790,7 @@ private long maybeUpdate(long now, Node node) {
         final RequestCompletionHandler callback;
         final boolean expectResponse;
         final boolean isInternalMetadataRequest; // used to flag metadata fetches which are triggered internally by NetworkClient
+        final boolean isInternalApiVersionRequest; // used to flag api version fetches which are triggered internally by NetworkClient
```

Comment:
We should probably have the `instanceof` checks anyway prior to casting the response object, so this just removes the need for an extra field. Is there any downside?